helper/keysutil/policy.go

```diff
@@ -836,7 +836,7 @@ func (p *Policy) HMACKey(version int) ([]byte, error) {
 	return p.Keys[strconv.Itoa(version)].HMACKey, nil
 }
 
-func (p *Policy) Sign(ver int, context, input []byte, algorithm string) (*SigningResult, error) {
+func (p *Policy) Sign(ver int, context, input []byte, hashalgorithm string, sigalgorithm string) (*SigningResult, error) {
```

Comment:
Specifying the type for hashAlgorithm can be skipped and `hashAlgorithm, sigAlgorithm string` should be sufficient.